sudoku_to_dag.py

```diff
@@ -0,0 +1,240 @@
+class Cell:
+    # set of possible values
+    values = []
+    # reference to the whole sudoku
+    sudoku = {}
+    # row and column indexes
+    row = 0
+    column = 0
+
+    final_value = 0
+
+    # initialize the cell with every possible value
+    def __init__(self, row, column, sudoku):
+        self.values = []
+        for i in range(1, sudoku.SIZE + 1):
+            self.values.append(i)
+        self.sudoku = sudoku
+        self.row = row
+        self.column = column
+
+    # returns the set of possible values for the cell
+    def get_possibilities(self):
+        return self.values
+
+    def remove_number(self, number):
+        try:
+            self.values.remove(number)
+            return True
+        except:
+            return False
+
+    # removes a possibility, if only one is remaining, calls the 'insert number' function,
+    # otherwise if there's no value left, it throws an exception
+    def remove_possibility(self, number):
+        if self.remove_number(number):
```

Comment:
Don't we also care about the else statement? I.e. if remove_number returns false we are looking at a "bad solution", non-solution, right?